### PR TITLE
Adds handling when re-scaffolding Dapr tasks

### LINKED ISSUE #33: Manage conflicts during scaffolding
Currently, if the target tasks/configurations created during scaffolding already exist, they will be silently overwritten. The extension should instead better manage those conflicts either by prompting to override and/or using different task/configuration names.

### PR DESCRIPTION
**Pull request by philliphoff**
Refactors the Dapr task scaffolding to detect when conflicts might occur (in either tasks or configurations) and allows the user to decide whether to overwrite a task/configuration or create a new one (e.g. when scaffolding for a second application in the repro).

> Note that these messages are modal rather than just notifications, as those will get hidden (after some time) and cause confusion when the user expects scaffolding to happen but doesn't, because it's waiting for the user to respond to the notification.

Resolves #33.

Screen Shot 2020-03-18 at 11 37 51 AM

Screen Shot 2020-03-18 at 11 38 32 AM

### CODE REVIEW

File: src/scaffolding/scaffolder.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                    break;
+                case 'skip':
+                    return undefined;
+            }
+        }
+
+        const content = await contentFactory(path);
+
+        await fse.writeFile(path, content, 'utf8');
+    }
```
Comment (ravipal) [inline src/scaffolding/scaffolder.ts]: Should this return the path?

File: src/commands/scaffoldDaprTasks.ts [hunk 2]
```diff
@@ -171,41 +185,95 @@ export async function scaffoldDaprTasks(context: IActionContext, ui: UserInput):
...
+            const overwrite: vscode.MessageItem = { title: localize('commands.scaffoldDaprTasks.overwriteTask', 'Overwrite') };
+            const newTask: vscode.MessageItem = { title: localize('commands.scaffoldDaprTasks.createTask', 'Create task') };
 
-    const daprdDownTask: DaprdDownTaskDefinition = {
-        type: 'daprd-down',
-        label: 'daprd-down',
-        appId: result.appId
-    };
+            const result = await ui.showWarningMessage(
+                localize('commands.scaffoldDaprTasks.taskExists', 'The task \'{0}\' already exists. Do you want to overwrite it or create a new task?', label),
```
[the remark this answers is not in the document]
Comment (philliphoff) [reply src/commands/scaffoldDaprTasks.ts]: I saw that feedback as well in your similar review, though tasks and, especially, configurations tend to have whitespace so it feels like the sentence could then be confusing.  (Filenames are less likely to have whitespace, so they tend to be more obvious in text.)

File: src/scaffolding/scaffolder.ts (hunk 1 above)
Comment (philliphoff) [reply src/scaffolding/scaffolder.ts]: Great catch, thanks!  I'm actually a little surprised and disappointed the TS compiler and/or linter failed to catch that.  Fixed.

File: src/commands/scaffoldDaprTasks.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ravipal) [reply src/commands/scaffoldDaprTasks.ts]: make sense.

**Pull request merged**